### DISCUSSION TITLE
An inner template with x-for is rendered twice although it should be only run once

### DISCUSSION THREAD

**Discussion by thebeautyofcoding** [alpinejs/alpine#3466] [7. Bugs (archived)]
Hi,

I am currently struggling a lot with the problem I described here:
https://stackoverflow.com/questions/75741373/laravel-alpinejs-expert-needed-for-strange-frontend-issue-a-nested-template-i

As you can see in the screenshots, sometimes the letterGroup, which is an array of letters (3 or 4) gets rendered twice. I get zero errors in the console and I am not seeing anything wrong with my AlpineJS Chrome extension. I am unable to grasp why this happens. I tried a lot of things for hours, but I keep getting this issue.

Thanks for the support :)

**Comment by SimoTod** [comment]
Do you have multiple copies of Alpine?
Can you replicate in a codepen?

**Reply by thebeautyofcoding** [reply]
Hello Mr Sir @SimoTod,

Thanks for your response 👍 Sorry for replying late, I had a 4 hour appointment and now I am just back. I think I do not have multiple copies but I will check it again.
I try to replicate it with codepen and let you know in a bit
Have a good one!

**Reply by thebeautyofcoding** [reply]
Hi @SimoTod,
For some reasons I am unable to get it working in codepen. Shame on me, I haven’t used it before. Unfortunately, I am super broken, because I barely got any sleep the last few days and 0 sleep last night, so I cannot concentrate anymore and have to drop to bed.
Someone posted a fiddle under my stackOverflow post: [jsfiddle.net/vk75439f](https://jsfiddle.net/vk75439f/)
This works flawlessly. However, he changed the JavaScript code a bit and deleted some more stuff from the blade template.
Maybe I am running AlpineJS code twice. This is what I have in my app.js and then in my blade template I had the 

    import './bootstrap'
    import Alpine from 'alpinejs'
    import focus from '@alpinejs/focus'
    
    
    window.Alpine = Alpine
    
    Alpine.plugin(focus)
    
    import keypad from './alpine/keypad'
    import browserCall from './alpine/browserCall'
    
    
    Alpine.data('keypad', keypad)
    Alpine.data('browserCall', browserCall)
    
    Alpine.start()
    

I have to look at it after I got some sleep. Sorry for that. Have a great day :)

**Reply by SimoTod** [reply]
Maybe check that you are not also importing the cdn file in the html template. That file looks okay.

**Reply by thebeautyofcoding** [reply]
What I found out is, as long as I keep commented out `this.$wire.set('number', this.value)`, I don't get this strange double render issue. I use `this.$wire` to set a property value in a Livewire component. This property reflects the value of the number input on top of the page. Any idea why this could cause this issue? Thanks a lot 👍
For more information, please check my stackOverflow post again.

**Reply by SimoTod** [reply]
Try to add wire:ignore to one of the parent divs

**Reply by chasegiunta** [reply]
Just came here to chime in that I'm experiencing the same issue. Super bizarre. Still investigating.

**Reply by gofish543** [reply]
Same issue. Still trying to fix

**Reply by ekwoka** [reply]
Livewire includes Alpine automatically, so you should not be separately adding it to the project.

**Reply by SpyridonLaz** [reply]
> Try to add wire:ignore to one of the parent divs

This worked for me even though i find it strange why livewire would iterate inside an x-for template when alpinejs is responsible for it.

**Comment by ngosangns** [comment]
Fixed. In my case, I have imported the Alpine.js library twice: once in Livewire and once manually. So I just remove one.
